docs/codeql/ql-language-reference/name-resolution.rst

```diff
@@ -164,14 +168,27 @@ As in many other programming languages, a namespace is a mapping from **keys** t
 **entities**. A key is a kind of identifier, for example a name, and a QL entity is
 a :ref:`module <modules>`, a :ref:`type <types>`, or a :ref:`predicate <predicates>`.
 
-Each module in QL has three namespaces:
+Each module in QL has six namespaces:
 
     - The **module namespace**, where the keys are module names and the entities are modules.
     - The **type namespace**, where the keys are type names and the entities are types.
     - The **predicate namespace**, where the keys are pairs of predicate names and arities, 
       and the entities are predicates.
+    - The **module signature namespace**, where the keys are module signature names and the entities are module signatures.
+    - The **type signature namespace**, where the keys are type signature names and the entities are type signatures.
+    - The **predicate signature namespace**, where the keys are pairs of predicate signature names and arities,
+      and the entities are predicate signatures.
 
-It's important to know that there is no relation between names in different namespaces. 
+The six namespaces of any module are not completely independent of each other:
```

Comment:
I think these should have a "the" in them e.g. `No keys may be shared between the **module namespace** and the **module signature namespace**` 

Potentially we also need to be clear that these ambiguity checks are between namespaces of a given module.

Reply:
I've added "the" in all those places with the third commit.

> Potentially we also need to be clear that these ambiguity checks are between namespaces of a given module.

Currently, my formulations are "The six namespaces **of any module** are not completely independent of each other:" and "**For each module**, some namespaces are enforced to be disjoint:". I could add something more elaborate if you have a suggestion?

Reply:
Not really. I think I am biased by my view of the code where we have group namespace then module rather than vice versa so per module becomes much less obvious.